integrations/lib/multiwoven/integrations/destination/zendesk/client.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+module Multiwoven
+  module Integrations
+    module Destination
+      module Zendesk
+        include Multiwoven::Integrations::Core
+        class Client < DestinationConnector
+          prepend Multiwoven::Integrations::Core::RateLimiter
+          def check_connection(connection_config)
+            connection_config = connection_config.with_indifferent_access
+            initialize_client(connection_config)
+            authenticate_client
+            success_status
+          rescue StandardError => e
+            handle_exception("ZENDESK:CHECK_CONNECTION:EXCEPTION", "error", e)
+            failure_status(e)
+          end
+
+          def discover(_connection_config = nil)
+            catalog = build_catalog(load_catalog)
+            catalog.to_multiwoven_message
+          rescue StandardError => e
+            handle_exception("ZENDESK:DISCOVER:EXCEPTION", "error", e)
+            failure_status(e)
+          end
+
+          def write(sync_config, records, action = "create")
+            @action = sync_config.stream.action || action
+            initialize_client(sync_config.destination.connection_specification)
+            process_records(records, sync_config.stream)
+          rescue StandardError => e
+            handle_exception("ZENDESK:WRITE:EXCEPTION", "error", e)
+            failure_status(e)
+          end
+
+          private
+
+          def initialize_client(connection_config)
+            connection_config = connection_config.with_indifferent_access
+            @client = ZendeskAPI::Client.new do |config|
+              config.url = ZENDESK_TICKETING_URL
+              config.username = connection_config[:username]
+              config.password = connection_config[:password]
```

Comment:
Ideally, it's advisable to use api token instead of password for security and maintainability reason especially when used in a third party application